cloud/cloud.go

```diff
@@ -81,7 +81,7 @@ type BatchManager interface {
 
 // GetManager returns an implementation of Manager for the given provider name.
 // It returns an error if the provider name doesn't have a known implementation.
-func GetManager(ctx context.Context, providerName string, settings *evergreen.Settings) (Manager, error) {
+func GetManager(ctx context.Context, providerName string, providerSettings *map[string]interface{}, settings *evergreen.Settings) (Manager, error) {
```

Comment:
the *map[string]interface{} is really a wart that I want to be able to get rid of at some point, so my thoughts here are: 

- remove provider name and provider settings and pass a host document? is there a situation where that wouldn't work? 
- convert the settings to whatever  type they should be (using mapstructure) and then pass them to this function as an interface, type-asserting them in the constructor. 
- if you only need some values, write a ManagerConfiguration data-struct with those options and pass that instead.

Reply:
I chose the third option -- there are some cases where we get a manager based on a Distro (not a Host) which made passing the Host document tricky. Besides, I needed to make a ManagerOpts struct for the grouping function below.